From: Ping Cheng <pinglinux@gmail.com>
Date: Sun, 24 Jun 2012 09:48:03 -0500
Subject: wacom: do not crash when retrieving touch_max
Bug-Debian: http://bugs.debian.org/678798

When rep_data was an array

	unsigned char rep_data[2];

spelling its address as &rep_data was perfectly valid, but now that
it is dynamically allocated

	unsigned char *rep_data = kmalloc(2, GFP_KERNEL);

that expression returns a pointer to the pointer rather than to the
array itself.  Regression introduced by commit f393ee2b814e (Input:
wacom - retrieve maximum number of touch points, 2012-04-29).

[jn: from mailing list discussion, with new description.
 This change is also available as part of a larger commit in the
 input-wacom repository.]

Signed-off-by: Ping Cheng <pingc@wacom.com>
Signed-off-by: Jonathan Nieder <jrnieder@gmail.com>
---
 drivers/input/tablet/wacom_sys.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/tablet/wacom_sys.c b/drivers/input/tablet/wacom_sys.c
index cad5602d3ce4..6b1cd71ba320 100644
--- a/drivers/input/tablet/wacom_sys.c
+++ b/drivers/input/tablet/wacom_sys.c
@@ -216,7 +216,7 @@ static void wacom_retrieve_report_data(struct usb_interface *intf,
 
 		rep_data[0] = 12;
 		result = wacom_get_report(intf, WAC_HID_FEATURE_REPORT,
-					  rep_data[0], &rep_data, 2,
+					  rep_data[0], rep_data, 2,
 					  WAC_MSG_RETRIES);
 
 		if (result >= 0 && rep_data[1] > 2)
-- 
1.7.11.rc3

